### PR TITLE
Log and notify for approvalnotes and source changes

### LINKED ISSUE #3913: Notify reviewers when "notes to reviewers" is updated
Reviewers often ask for additional information which developers provide using the _Notes to Reviewers_ field.
However, reviewers have no way of knowing when that field was updated.

If the _info-request_ flag is set and the _Notes to Reviewers_ field is updated, we should clear the flag and send an email to the reviewer. Also, an event like `Notes to Reviewers have been updated` should be added to the review activity.

If the _info-request_ flag is *not* set, we don't need to sent emails, I think. Adding the event wouldn't hurt though.

Adding @kewisch as this might be relevant to him.

### PR DESCRIPTION
**Pull request by wagnerand**
Fixes #3913.

### CODE REVIEW

**Comment by wagnerand** [comment]
Unrelated test fail?

**Comment by wagnerand** [comment]
@eviljeff this should be good for r? now

There are a couple of behavior changes, for example we will only log and notify source attachments if the version was pending information. There will be no logging for attaching source code if it is attached during submission of the xpi.
This does not affect setting the admin-flag.

Adding @jvillalobos so he's aware.

File: src/olympia/devhub/views.py [hunk 1]
```diff
@@ -1166,11 +1167,23 @@ def version_edit(request, addon_id, addon, version_id):
...
 
-        fields = ('source', 'approvalnotes')
-        has_changed = [field in version_form.changed_data for field in fields]
-        if version.has_info_request and any(has_changed):
-            version.update(has_info_request=False)
-            version.save()
+        if 'approvalnotes' in version_form.changed_data:
+            if version.has_info_request:
+                version.update(has_info_request=False)
+                log_and_notify(amo.LOG.APPROVALNOTES_CHANGED,
```
Comment (eviljeff) [inline src/olympia/devhub/views.py]: You need to put the use of `log_and_notify` behind the `activity-email` waffle switch for now.  I'd just create the log() if the waffle is off.

File: src/olympia/devhub/views.py [hunk 2]
```diff
@@ -1166,11 +1167,23 @@ def version_edit(request, addon_id, addon, version_id):
...
+                               _('Approval notes were updated.'),
+                               request.user,
+                               version)
+
+        if 'source' in version_form.changed_data:
+            addon.update(admin_review=True)
+            if version.has_info_request:
+                version.update(has_info_request=False)
+                log_and_notify(amo.LOG.SOURCECODE_UPLOADED,
```
Comment (eviljeff) [inline src/olympia/devhub/views.py]: same with this one - needs waffling.

File: src/olympia/amo/log.py [hunk 3]
```diff
@@ -671,6 +671,21 @@ class REVIEWER_REPLY_VERSION(_LOG):
     review_queue = True
 
 
+class APPROVALNOTES_CHANGED(_LOG):
```
Comment (eviljeff) [inline src/olympia/amo/log.py]: you might want to add this to https://github.com/mozilla/addons-server/blob/master/src/olympia/activity/utils.py#L228 - those are the events that make notes before them 'read'.

Comment (wagnerand) [inline src/olympia/amo/log.py]: Yeah I did not get what that list was about. So, should I add both `APPROVALNOTES_CHANGED` and `SOURCECODE_UPLOADED`?

Comment (eviljeff) [inline src/olympia/amo/log.py]: `APPROVALNOTES_CHANGED` yes; `SOURCECODE_UPLOADED` probably.

Comment (eviljeff) [inline src/olympia/amo/log.py]: `APPROVAL_NOTES_CHANGED` and `SOURCE_CODE_UPLOADED` is more consistent naming

**Comment by wagnerand** [comment]
Updated. Tests should already cover the code since I just replaced the version watcher by setting the admin flag properly.

**Comment by wagnerand** [comment]
😞 more test failures...

File: src/olympia/devhub/tests/test_views_versions.py [hunk 4]
```diff
@@ -821,12 +821,6 @@ def test_should_accept_zip_source_file(self):
         assert version.source
         assert version.addon.admin_review
 
-        # Check that the corresponding automatic activity log has been created.
-        log = ActivityLog.objects.get(action=amo.LOG.REQUEST_SUPER_REVIEW.id)
```
Comment (eviljeff) [inline src/olympia/devhub/tests/test_views_versions.py]: check for `SOURCE_CODE_UPLOADED` instead

File: src/olympia/devhub/views.py [hunk 5]
```diff
@@ -1166,11 +1167,31 @@ def version_edit(request, addon_id, addon, version_id):
...
+        if 'approvalnotes' in version_form.changed_data:
+            if version.has_info_request:
+                version.update(has_info_request=False)
+                if waffle.switch_is_active('activity-email'):
+                    log_and_notify(amo.LOG.APPROVAL_NOTES_CHANGED,
+                                   _('Approval notes were updated.'),
+                                   request.user,
+                                   version)
+                else:
+                    amo.log(amo.LOG.APPROVAL_NOTES_CHANGED,
```
Comment (eviljeff) [inline src/olympia/devhub/views.py]: Add a check we logged the note in `test_update_approvalnotes_should_drop_info_request_flag`

**Pull request merged**